nodes.go

```diff
@@ -184,6 +202,40 @@ type group struct {
 	mode groupMatchMode
 }
 
+func (g *group) Fuzz(l lexer.Fuzzer) string {
+	var (
+		maxCount int
+		minCount int
+	)
+
+	switch g.mode {
+	case groupMatchOnce:
+		minCount, maxCount = 1, 1
+	case groupMatchZeroOrOne:
+		minCount, maxCount = 0, 1
+	case groupMatchZeroOrMore:
+		minCount, maxCount = 0, math.MaxInt
+	case groupMatchOneOrMore:
+		minCount, maxCount = 1, math.MaxInt
+	case groupMatchNonEmpty:
+		minCount, maxCount = 1, 1
+	}
+
+	var items int
+	if maxCount-minCount == 0 {
+		items = minCount
+	} else {
+		items = minCount + rand.Intn(maxCount-minCount)
+	}
+	var s strings.Builder
+	for i := 0; i < items; i++ {
+		s.WriteString(g.expr.Fuzz(l))
+		if i < items-1 {
+			s.WriteString(" ")
```

Comment:
This is a _really_ interesting idea, but this implementation makes assumptions about whitespace that will not hold true for all lexers. For example, it is possible to have a lexer that relies on indentation, which this would break on.

I love the idea though, so perhaps there is a way to make it work in the general case.

Reply:
> This is a really interesting idea, but this implementation makes assumptions about whitespace that will not hold true for all lexers. For example, it is possible to have a lexer that relies on indentation, which this would break on.

Hm, how would you model this? I was trying to keep it stateless, but I'm not sure if I could avoid that and fix this problem.

Reply:
Yes, that is definitely the question, and I'm not sure what the answer is :(